v3/src/components/graph/components/binneddotplotdots.tsx

```diff
@@ -151,23 +236,52 @@ export const BinnedDotPlotDots = observer(function BinnedDotPlotDots(props: Plot
       pointDisplayType, anchor: circleAnchor
     })
   },
-  [graphModel, dataConfig, layout, primaryAttrRole, secondaryAttrRole, dataset, pixiPoints,
-    primaryIsBottom, pointColor, pointStrokeColor, isAnimating, pointDisplayType])
+  [dataConfig, primaryIsBottom, graphModel, primaryPlace, layout, primaryAttrRole, secondaryAttrRole,
+   drawBinBoundaries, dataset, primaryAxisScale, pixiPoints, pointColor, pointStrokeColor, isAnimating,
+   pointDisplayType, addBinBoundaryDragHandlers])
 
   usePlotResponders({pixiPoints, refreshPointPositions, refreshPointSelection})
 
-  // respond to binAlignment and binWidth changes because the axis may need to be updated
+  // Respond to binAlignment and binWidth changes because the axis may need to be updated.
   useEffect(function respondToGraphBinSettings() {
     return mstReaction(
-      () => [graphModel.binAlignment, graphModel.binWidth],
+      () => [graphModel._binAlignment, graphModel._binWidth, graphModel.binAlignment, graphModel.binWidth],
       () => {
       refreshPointPositions(false)
     }, {name: "respondToGraphBinSettings"}, graphModel)
-  }, [dataset, graphModel, refreshPointPositions])
+  }, [addBinBoundaryDragHandlers, dataset, graphModel, refreshPointPositions])
+
+  // Initialize binWidth and binAlignment on the graph model if they haven't been defined yet.
+  // This can happen when a CODAP document containing a graph with binned points is imported.
+  useEffect(function setInitialBinSettings() {
+    if (!dataConfig) return
+    if (graphModel.binWidth === undefined || graphModel.binAlignment === undefined) {
+      const { binAlignment, binWidth } = dataConfig.binDetails()
+      graphModel.setBinWidth(binWidth)
+      graphModel.setBinAlignment(binAlignment)
+    }
+  })
+
+  // If the pixel width of graphModel.binWidth would be less than kMinBinPixelWidth, set it to kMinBinPixelWidth.
+  useEffect(function enforceMinBinPixelWidth() {
+    return mstReaction(
+      () => [graphModel.binWidth, primaryAxisScale],
+      () => {
+        if (graphModel.binWidth && primaryAxisScale) {
+          const pixelWidth = primaryAxisScale(graphModel.binWidth) - primaryAxisScale(0)
+          if (pixelWidth < kMinBinPixelWidth) {
+            const newBinWidth = primaryAxisScale.invert(primaryAxisScale(0) + kMinBinPixelWidth) - primaryAxisScale(0)
+            graphModel.setBinWidth(newBinWidth)
+          }
+        }
+      }, {name: "enforceMinBinPixelWidth"}, graphModel
+    )
+  }, [graphModel, primaryAxisScale])
 
   return (
-    <>
-      <g data-testid={`bin-ticks-${instanceId}`} className="bin-ticks" ref={binTicksRef}/>
-    </>
+    plotArea2Ref?.current && createPortal(
+      <><g data-testid={`bin-ticks-${instanceId}`} className="bin-ticks" ref={binBoundariesRef}/></>,
+      plotArea2Ref.current
+    )
```

Comment:
I would suggest using Chakra's `<Portal>` here, which should behave the same but make the code a bit tidier. Also probably worth a comment about the fact that we're portaling to put the user-manipulable bin boundaries above the points.

Reply:
[This was resolved in an off-Github conversation where we determined we would stick with using `createPortal` for now since the target is an SVG `g` element, and Chakra's `Portal` requires an HTML element.]